### ISSUE TITLE
ERROR ITMS-90060 (CFBundleShortVersionString)

### ISSUE THREAD

**Issue by lifeisfoo** [bustoutsolutions/siesta#24]
Trying to upload the ipa to the app store I receive this error:
> ERROR ITMS-90060: "This bundle is invalid. The value for key CFBundleShortVersionString '1.0-beta.4' in the Info.plist file must be a period-separated list of at most three non-negative integers."

Seems to be a [very](https://github.com/square/SocketRocket/issues/302) [common](https://github.com/CocoaPods/CocoaPods/issues/4421) [problem](https://github.com/Carthage/Carthage/issues/859), since from some time Apple is enforcing a _three non-negative integers_ version number even for modules/frameworks.

There is also a dedicated document from Apple [about version and build numbers](https://developer.apple.com/library/ios/technotes/tn2420/_index.html#//apple_ref/doc/uid/DTS40016603-CH1-TWO_NAMING_CONVENTIONS_FOR_VERSION_NUMBERS_AND_BUILD_NUMBERS)

**Comment by pcantrell** [comment]
Thanks for the heads up. Not sure how this crept into an info plist! Are you building with Cocoapods, Carthage, or manual code inclusion (e.g. git submodule)?

Would you mind giving me a dump of everywhere the string appears in your codebase? That might reveal something:

    find . -name .git -prune -o -type f -exec grep -l 1.0-beta

(Also, note that beta 5 just came out!)

**Comment by lifeisfoo** [comment]
I'm using 1.0-beta4 with Cocoapods.

These files contains the version number:
```
./Podfile
./Podfile.lock
./Pods/Manifest.lock
./Pods/Siesta/README.md
./Pods/Target Support Files/Siesta/Info.plist
```

I fixed it editing the version number in the last file (now .ipa upload works fine).

**Comment by pcantrell** [comment]
Ah — so Cocoapods is generating a plist and stuffing it in. One of my Siesta apps uses Cocoapods too. Wonder why this issue didn’t trip that one up? Curious. Would you mind attaching your Podfile (if it's not full of secret stuff)?

**Comment by lifeisfoo** [comment]
Sure.

Podfile:
```
platform :ios, '8.0'
use_frameworks!

pod 'Siesta', '~>1.0-beta.4'
pod 'SwiftyJSON', '~> 2.3'

target 'MyApp' do

end
```

Podfile.lock:
```
PODS:
  - Siesta (1.0-beta.4):
    - Siesta/Core (= 1.0-beta.4)
  - Siesta/Core (1.0-beta.4)
  - SwiftyJSON (2.3.2)

DEPENDENCIES:
  - Siesta (~> 1.0-beta.4)
  - SwiftyJSON (~> 2.3)

SPEC CHECKSUMS:
  Siesta: 4bda1f6e7e0431b64bdfe9d3f902c34d90822c27
  SwiftyJSON: 04ccea08915aa0109039157c7974cf0298da292a

COCOAPODS: 0.39.0
```

**Comment by pcantrell** [comment]
Yes, confirmed locally, and it is indeed an issue in Cocoapods, the one that you linked to in your first comment: https://github.com/CocoaPods/CocoaPods/issues/4421

From what segeddins said on that thread, it looks like there’s already a fix in the pipelines, and it may be released as soon as Cocoapods 0.39.1. In the meantime, you could try the `post_install` hacks on that thread — but it may be easier to just manually fix the version number until the updated Cocoapods is released.

> One of my Siesta apps uses Cocoapods too. Wonder why this issue didn’t trip that one up?

Because the only release of that particular app that I did _after_ the Siesta beta numbering started used a hot patch from master, and thus wasn’t stamped with a beta version number. So that explains it.

What’s your app, if you don’t mind my asking? Would love to see it!
